plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/templateFile.ts

```diff
@@ -0,0 +1,167 @@
+ * Then places the result in the workspace, or optionally in a subdirectory
+ * specified by the 'targetPath' input option.
+ * @public
+ */
+export function createFetchTemplateFileAction(options: {
+  reader: UrlReaderService;
+  integrations: ScmIntegrations;
+  additionalTemplateFilters?: Record<string, TemplateFilter>;
+  additionalTemplateGlobals?: Record<string, TemplateGlobal>;
+}) {
+  const {
+    reader,
+    integrations,
+    additionalTemplateFilters,
+    additionalTemplateGlobals,
+  } = options;
+
+  const defaultTemplateFilters = createDefaultFilters({ integrations });
+
+  return createTemplateAction<{
+    url: string;
+    targetPath: string;
+    values: any;
+    cookiecutterCompat?: boolean;
+    replace?: boolean;
+    trimBlocks?: boolean;
+    lstripBlocks?: boolean;
+    token?: string;
+  }>({
+    id: 'fetch:template:file',
+    description: 'Downloads single file and places it in the workspace.',
+    examples,
+    schema: {
+      input: {
+        type: 'object',
+        required: ['url', 'targetPath'],
+        properties: {
+          url: {
+            title: 'Fetch URL',
+            description:
+              'Relative path or absolute URL pointing to the single file to fetch.',
+            type: 'string',
+          },
+          targetPath: {
+            title: 'Target Path',
+            description:
+              'Target path within the working directory to download the file as.',
+            type: 'string',
+          },
+          values: {
+            title: 'Template Values',
+            description: 'Values to pass on to the templating engine',
+            type: 'object',
+          },
+          cookiecutterCompat: {
+            title: 'Cookiecutter compatibility mode',
+            description:
+              'Enable features to maximise compatibility with templates built for fetch:cookiecutter',
+            type: 'boolean',
+          },
+          replace: {
+            title: 'Replace file',
+            description:
+              'If set, replace file in targetPath instead of overwriting existing one.',
+            type: 'boolean',
+          },
+          token: {
+            title: 'Token',
+            description:
+              'An optional token to use for authentication when reading the resources.',
+            type: 'string',
+          },
+        },
+      },
+    },
+    supportsDryRun: true,
+    async handler(ctx) {
+      ctx.logger.info('Fetching template file content from remote URL');
+
+      const workDir = await ctx.createTemporaryDirectory();
+      // Write to a tmp file, render the template, then copy to workspace.
+      const tmpFilePath = path.join(workDir, 'tmp');
+
+      const outputPath = resolveSafeChildPath(
+        ctx.workspacePath,
+        ctx.input.targetPath,
+      );
+
+      if (fs.existsSync(outputPath) && !ctx.input.replace) {
+        ctx.logger.info('File already exists in workspace, not replacing.');
```

Comment:
Maybe worth mentioning which file exactly exists?
As the end user you have no idea what a template developer configured that file to be 😅

Reply:
Done